api/src/main/java/edu/cornell/mannlib/vitro/webapp/auth/RootUserSetup.java

```diff
@@ -0,0 +1,182 @@
+/* $This file is distributed under the terms of the license in LICENSE$ */
+
+package edu.cornell.mannlib.vitro.webapp.auth;
+
+import java.util.TreeSet;
+
+import javax.servlet.ServletContext;
+import javax.servlet.ServletContextEvent;
+import javax.servlet.ServletContextListener;
+
+import edu.cornell.mannlib.vitro.webapp.beans.UserAccount;
+import edu.cornell.mannlib.vitro.webapp.beans.UserAccount.Status;
+import edu.cornell.mannlib.vitro.webapp.config.ConfigurationProperties;
+import edu.cornell.mannlib.vitro.webapp.controller.authenticate.Authenticator;
+import edu.cornell.mannlib.vitro.webapp.dao.UserAccountsDao;
+import edu.cornell.mannlib.vitro.webapp.modelaccess.ModelAccess;
+import edu.cornell.mannlib.vitro.webapp.startup.StartupStatus;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+/**
+ *
+ * On setup, check to see that the specified root user exists. If not, create it. If we can't create it, abort.
+ *
+ * If any other root users exist, warn about them.
+ */
+public class RootUserSetup implements ServletContextListener {
```

Comment:
As experiment with moving HasProfileOrIsBlacklistedFactory didn't show any effect, I suggest to leave it like it is now.

Reply:
ok, we can keep it as is, if someone in interested in the history,  it can be found in the another class. It would be great if you can keep in your mind to use git mv in future PR when there is a sense for that